pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -572,7 +573,7 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       // Decode message
       StreamDataDecoderResult decodedRow = _streamDataDecoder.decode(messagesAndOffsets.getStreamMessage(index));
       msgMetadata = messagesAndOffsets.getStreamMessage(index).getMetadata();
-      StreamPartitionMsgOffset messageOffset = messagesAndOffsets.getNextStreamPartitionMsgOffsetAtIndex(index);
+      int messageOffset = messagesAndOffsets.getMessageOffsetAtIndex(index);
```

Comment:
Does this give the accurate offset? Will it cause overflow?

Reply:
I don't see it marked as deprecated and there is nothing similar to `getNextStreamPartitionMsgOffsetAtIndex` which returns StreamPartitionMsgOffset (it has long value).

Will wait for @satishd to chime in here as he might have more context on this.

Reply:
[MessageBatch](https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java#L73)'s `getMessageOffsetAtIndex(index)` javadoc says it gives the offset of the message at the given index. But the respective Kafka implementation does not seem to return that. What does that method really mean and what is it used for?

We need Kafka message's record offset here. So, we need a similar implementation like `getNextStreamPartitionMsgOffsetAtIndex()` to get it from the metadata properties and get the message record's offset.

Reply:
@satishd gave a shot to having a `getStreamPartitionMsgOffsetAtIndex` in MessageBatch. Let me know if this looks good.